File: Permutation Sequence.java
// Leetcode Problem 60: Permutation Sequence
// JAVA CODE
class Solution {
    public String getPermutation(int n, int k) {
        List<Integer> numbers = new ArrayList<>();
        int[] factorial = new int[n + 1];
        factorial[0] = 1;

        for (int i = 1; i <= n; i++) {
            numbers.add(i);
            factorial[i] = factorial[i - 1] * i;
        }

        k--;
        StringBuilder result = new StringBuilder();

        for (int i = n; i >= 1; i--) {
            int index = k / factorial[i - 1];
            result.append(numbers.get(index));
            numbers.remove(index);
            k %= factorial[i - 1];
        }

        return result.toString();
    }
}

File: Permutation Sequence.cpp
// Leetcode Problem 60: Permutation Sequence
// C++ CODE
class Solution {
public:
    string getPermutation(int n, int k) {
        vector<int> numbers;
        vector<int> factorial(n + 1, 1);

        for (int i = 1; i <= n; i++) {
            numbers.push_back(i);
            factorial[i] = factorial[i - 1] * i;
        }

        k--;
        string result = "";

        for (int i = n; i >= 1; i--) {
            int index = k / factorial[i - 1];
            result += to_string(numbers[index]);
            numbers.erase(numbers.begin() + index);
            k %= factorial[i - 1];
        }

        return result;
    }
};